schema.ts

```diff
@@ -1303,15 +1303,51 @@ export interface AnyonePrincipal {
 type Principal = UserPrincipal | GroupPrincipal | AnyonePrincipal;
 
 /**
- * This represents the definition of a permission in the external system.
+ * The type of permission that is being granted
  *
  * TODO(sam): Unhide this
  * @hidden
  */
-export interface Permission {
+export enum PermissionType {
+  /**
+   * Represents a permission a user, group, or anyone has to view an item.
+   */
+  Principal = 'principal',
+  /**
+   * Represents a permission that delegates access to permissions on a separate entity.
+   */
+  Delegated = 'delegated',
+}
+
+/**
+ * This represents a permission that is granted to a principal.
+ *
+ * TODO(sam): Unhide this
+ * @hidden
+ */
+export interface PrincipalPermission {
+  type: PermissionType.Principal;
   principal: Principal;
 }
 
+/**
+ * This represents a permission that delegates access to another entity
```

Comment:
Another entity of the same type, I assume? I don't see a way here to specify a foreign type.

Reply:
Yeah this would be of the same time as the parent.